lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -514,7 +515,14 @@ export class UserAgentApplication {
             }
         }
 
-        acquireTokenAuthority.resolveEndpointsAsync(this.telemetryManager, request.correlationId).then(async () => {
+        try {
+            if (!acquireTokenAuthority.hasCachedMetadata()) {
```

Comment:
Instead of checking for this here, why don't we just do this inside resolveAuthorityAsync()? If the cached metadata is available, just return it, otherwise make a network request (similar to acquireTokenSilent)

Reply:
I like this too. Though this would prevent intentional refreshes without an additional `forceRefresh` parameter. Not sure if there's a valid argument for supporting that use case but something to consider.

Reply:
We can't do intentional refreshes anyway with the current code.

Reply:
One of the goals is to minimize the number of async code paths, which is why the `if` is done here, so that way if the check fails, we don't enter an async function. If this was factored out, it would entered the async every time.

Reply:
Also, I did not consider if `forceRefresh` should force a network call for openid configuration, in addition to forcing a network call for a new token. Initially, I would think not. @DarylThayil any opinion here?